### ISSUE TITLE
30-second timeout when launching any app

### ISSUE THREAD

**Issue by jurf** [flatpak/flatpak#3223]
## Linux distribution and version
F30–F31.

## Flatpak version
1.4.3

## Description of the problem
When launching any app, Flatpak waits for 30 seconds after `F: Add locks in dir…`.

**Comment by jurf** [comment]
Should mention that there is no CPU nor disk activity during the wait. The Fedora installation is, apart from some minor changes, completely standard.

**Comment by jurf** [comment]
Well, turns out removing `xdg-desktop-portal-gtk` fixes the issue, but that’s not really a solution.

**Comment by refi64** [comment]
Could it be a timeout trying to talk to the portal for something? Is there
anything in the journal around the time this occurs?

**Comment by jurf** [comment]
Hm.
```
lis 15 21:39:25 argentum polkitd[934]: Registered Authentication Agent for unix-process:7811:1401105 (system bus name :1.1015 [flatpak run org.telegram.desktop], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale cs_CZ.UTF-8)
lis 15 21:39:25 argentum systemd[2405]: Starting flatpak document portal service...
lis 15 21:39:25 argentum xdg-document-portal[7818]: error: Failed to load db: invalid gvdb header
lis 15 21:39:25 argentum systemd[2405]: xdg-document-portal.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
lis 15 21:39:25 argentum systemd[2405]: xdg-document-portal.service: Failed with result 'exit-code'.
lis 15 21:39:25 argentum systemd[2405]: Failed to start flatpak document portal service.
lis 15 21:39:39 argentum xdg-desktop-por[3319]: Failed to get application states: GDBus.Error:org.freedesktop.portal.Error.Failed: Could not get window list: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: App introspection not allowed
```

The `xdg-document-portal[7818]: error: Failed to load db: invalid gvdb header` looks promising.

**Comment by jurf** [comment]
A quick search makes it look like there’s a corrupted dconf database somewhere?

**Comment by jurf** [comment]
Yep, `~/.local/share/flatpak/db/documents` caused the issue. Should I close this or is the corruption worth investigating?

**Comment by jurf** [comment]
…the file was _empty_? Does that make any sense?

**Comment by refi64** [comment]
What if you try straight up deleting the file?

**Comment by jurf** [comment]
I moved the entire folder away, no issue since then. Also, I guess the empty file makes sense – the error was because it couldn’t find the header, which definitely wasn’t in the empty file. The question is why it got there however.

**Comment by refi64** [comment]
The document portal writes the permission database in an atomic fashion, so
afaik the only real way this would break would be either a filesystem issue
or a bizarre gvdb issue. Either way, it's arguably a random fluke that at
minimum would be quite hard to reproduce.
